src/frequenz/client/reporting/_client.py

```diff
@@ -6,23 +6,19 @@
 from collections import namedtuple
 from dataclasses import dataclass
 from datetime import datetime
-from typing import Any, AsyncIterator, Awaitable, Iterator, Type, cast
+from typing import Any, AsyncIterator, Iterator, Type, cast
```

Comment:
Nitpick, but those classes were added to the `typing` module when the type hints system wasn't mature enough and regular classes couldn't be used for typing I think. They are deprecated in typing since Python 3.9.

```suggestion
from typing import Any, Type, cast
from collections.abc import AsyncIterator, Iterator
```

![image](https://github.com/frequenz-floss/frequenz-client-reporting-python/assets/1031485/2e11a2e6-f4a7-4cef-a625-c6fc51b9e2ca)

Reply:
Fixed